compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingTypes.h

```diff
@@ -63,6 +63,29 @@ MatmulNarrowDim getMatmulNarrowDim(linalg::LinalgOp linalgOp,
                                    int narrowThreshold);
 
 // The structs defined here because they are used by encoding_interfaces.td.
+
+/// Bundles an encoding attribute with its associated dynamic information.
+/// This is returned by interface methods that query encoding properties
+/// from operations. Dynamic values can include runtime information needed
+/// by the encoding (e.g., M, N, K dimensions for matmul encodings).
+struct EncodingProperties {
+  /// The encoding attribute for the operand/result.
+  Attribute encoding;
+  /// Dynamic values needed by the encoding. For matmul-like operations,
+  /// these typically correspond to the iteration domain dimensions (M, N, K).
+  /// TODO(jtuyls): Currently empty; will be populated to support dynamic
+  /// layout decisions. See https://github.com/iree-org/iree/issues/22370.
```

Comment:
Linking with github issue directly looks better to me. IMO, we use handle when the corresponding issue is not present. It means that asking this person who has the context to me. See https://google.github.io/styleguide/cppguide.html#TODO_Comments


```suggestion
  /// TODO(#22370): Currently empty; will be populated to support dynamic
  /// layout decisions.
```